service_grpc_flight/src/lib.rs

```diff
@@ -486,7 +534,70 @@ mod tests {
         let read_info = ReadInfo::decode_json(&ticket.ticket).unwrap();
 
         assert_eq!(read_info.namespace_name, "my_db");
-        assert_eq!(read_info.sql_query, "SELECT 1;");
+        assert_matches!(read_info.query, Query::Sql(query) => assert_eq!(query, "SELECT 1;"));
+    }
+
+    #[test]
+    fn test_read_info_decoding() {
+        let mut buf = Vec::with_capacity(1024);
+        proto::ReadInfo::encode(
+            &proto::ReadInfo {
+                namespace_name: "<foo>_<bar>".to_string(),
+                sql_query: "SELECT 1".to_string(),
+                query_type: QueryType::Unspecified as i32,
```

Comment:
You can write this slighty more idomatically like

```suggestion
                query_type: QueryType::Unspecified.into(),
```